backend/overlapping-block-marker/Marker.py

```diff
@@ -0,0 +1,159 @@
+
+        # Translate category number to category
+        category = converter.get_category(category_n)
+
+        # Assign it to the block
+        block.assigned_category = category
+        block.annotation = converter.annotation_map[category]
+        return block
+
+    def calculate_total_score(self, source_blocks_map):
+        
+        tot_blocks = 0
+        tot_ok = 0
+        
+        for source in self.sources:
+            blocks = source_blocks_map[source]
+            tot_blocks += len(blocks)
+            for block in blocks:
+                annotation = block.annotation
+                if (annotation != '' and
+                    (annotation == 'ok' or annotation == 'comment')):
+                    tot_ok += 1
+        if tot_blocks == 0:
+            return 0
+        return (float(tot_ok)/float(tot_blocks)) * 100
+    
+    def generate_annotations_output(self, source_blocks_map):
+        
+        all_annotations = []
+        
+        # Generate Annotations
+        for source in self.sources:
+            blocks = source_blocks_map[source]
+            for block in blocks:
+                ann = block.annotation
+                if (ann != '' and ann != 'ok' and ann != 'comment'):
+                    e_type = self.category_converter.error_map[block.assigned_category]
+                    annotation = Annotation(
+                                    ann, source,
+                                    block.lineno, 
+                                    block.charno,
+                                    e_type)
+                    all_annotations.append(annotation.get_json())
+        return all_annotations
+
+    # Generates json output for the marker
+    def generate_json(self, source_blocks_map, error):
+        
+        annotations = self.generate_annotations_output(source_blocks_map)
+        score = self.calculate_total_score(source_blocks_map)
+        json_output = {
+            'score': score,
+            'annotations': annotations,
+            'error': error
+        }
+        return json_output
+
+    def write_output(self, output_f):
+        if self.output != None:
+            _file = open(output_f, 'w')
+            json.dump(self.output, _file)
+            _file.close()
+
+    def mark(self):
+        
+        # Holds mapping between source and file's blocks
+        partial_outputs = {}
+        error = None
+        
+
+        # Iterate through files to classify
+        for source in self.sources:
+    
+            # Get blocks
+            splitter = Script_Blocks_Container(source, self.sample_size)
+            splitter.split()
+            blocks = splitter.container
+        
+            to_classify = []
+            for block in blocks:
+                block.format_content()
+                to_classify.append(block.formatted_content)
+            
+            guesses = []
+            # Classify
+            try:
+                guesses = self.classifier.predict(to_classify)
```

Comment:
[...] all the predicted *categories associated to the blocks. I will describe the way this all works in the description, maybe that's better.